src/RadixSortLSDClass.chpl

```diff
@@ -0,0 +1,225 @@
+                            }
+                            aggregator.flush();
+                        }//coforall task
+                    }//on loc
+                }//coforall loc
+
+                // scan globalCounts to get bucket ends on each locale/task
+                var globalStarts = + scan globalCounts;
+                globalStarts -= globalCounts;
+
+                try! rsLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                                    "globalCounts = %t".format(globalCounts));
+                try! rsLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                                    "globalStarts = %t".format(globalStarts));
+
+                // calc new positions and permute
+                coforall loc in Locales {
+                    on loc {
+                        coforall task in Tasks {
+                            // bucket domain
+                            var bD = {0..#numBuckets};
+                            // allocate counts
+                            var taskBucketPos: [bD] int;
+                            // get local domain's indices
+                            var lD = aD.localSubdomain();
+                            // calc task's indices from local domain's indices
+                            var tD = calcBlock(task, lD.low, lD.high);
+                            // read start pos in to globalStarts back from transposed order
+                            {
+                                var aggregator = newSrcAggregator(int);
+                                for bucket in bD {
+                                    aggregator.copy(taskBucketPos[bucket],
+                                            globalStarts[calcGlobalIndex(bucket, loc.id, task)]);
+                                }
+                                aggregator.flush();
+                            }
+                            // calc new position and put data there in temp
+                            {
+                                var aggregator = newDstAggregator(t);
+                                for i in tD {
+                                    const ref tempi = temp.localAccess[i];
+                                    const key = comparator.key(tempi);
+                                    var bucket = getDigit(key, rshift, last, negs); // calc bucket from key
+                                    var pos = taskBucketPos[bucket];
+                                    taskBucketPos[bucket] += 1;
+                                    aggregator.copy(a[pos], tempi);
+                                }
+                                aggregator.flush();
+                            }
+                        }//coforall task
+                    }//on loc
+                }//coforall loc
+
+                // copy back to temp for next iteration
+                // Only do this if there are more digits left
+                if !last {
+                temp = a;
```

Comment:
we just accepted a PR to use the swap operator `temp <=> a`, need to reflect the change here.